hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/cost/Cost.java

```diff
@@ -36,7 +36,7 @@ public class Cost implements RelOptCost {
 
     public static final Cost ZERO = new Cost(0.0d, 0.0d, 0.0d);
     public static final Cost TINY = new Cost(1.0d, 1.0d, 0.0d);
-    public static final Cost HUGE = new Cost(Double.MAX_VALUE, Double.MAX_VALUE, Double.MAX_VALUE);
+    public static final Cost HUGE = new Cost(Float.MAX_VALUE, Float.MAX_VALUE, Float.MAX_VALUE);
```

Comment:
`Double.MAX_VALUE / 2` should cause less head scratching

Reply:
But is it actually a case? Float.MAX_VALUE seems like a good candidate to be `HUGE` cost.

Reply:
you rely on the assumption that Float.MAX_VALUE  < Double.MAX_VALUE, which usually is true, by why rely on that if you can be explicit with `Double.MAX_VALUE / 2` - less surprise and thinking for the reader.

Reply:
https://github.com/hazelcast/hazelcast/pull/24199/commits/f21236024d87fd1851ed834460b46512df295286